### PR TITLE
MGMT-8765: Add integration tests to the MGMT-8280 code

### PR DESCRIPTION
**Pull request by isaacdorfman**
# Assisted Pull Request

## Description
The PR for the ticket [MGMT-8280](https://issues.redhat.com/browse/MGMT-8280) did not include integration tests.
This PR adds them to the code.

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [x] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [x] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: subsystem/kubeapi_test.go [hunk 1]
```diff
@@ -881,73 +872,74 @@ var _ = Describe("[kube-api]cluster installation", func() {
...
 		Eventually(func() bool {
-			aci := &hiveext.AgentClusterInstall{}
-			err := kubeClient.Get(ctx, installkey, aci)
-			return apierrors.IsNotFound(err)
-		}, "30s", "10s").Should(Equal(true))
-		By("Verify ClusterDeployment Agents were deleted")
-		Eventually(func() int {
-			return len(getClusterDeploymentAgents(ctx, kubeClient, clusterKey).Items)
-		}, "2m", "2s").Should(Equal(0))
+			dbCluster, err := common.GetClusterFromDBWhere(db, common.UseEagerLoading, common.SkipDeletedRecords, "kube_key_name = ? and kube_key_namespace = ?", clusterkey.Name, clusterkey.Namespace)
```
Comment (eranco74) [inline subsystem/kubeapi_test.go:920]: Try with `common.IncludeDeletedRecords`

**Comment by ybettan** [comment]
Can we please have a better commit message including its title? (including the PR description and title).
Also can you please squash all your commits? Otherwise, github will squash it for you and the commit message will be something like:
```
* MGMT-8765: Add integration tests to the MGMT-8280 code
* MGMT-8765: WIP
* Merge branch 'master' into MGMT-8765
* MGMT-8765: Fixing tests
* Merge branch 'master' into MGMT-8765
* MGMT-8765: Fixing tests
* MGMT-8765: Fixing tests
* MGMT-8765: Fixing tests
```

Which doesn't describe the actual content of the commit.

File: subsystem/kubeapi_test.go [hunk 2]
```diff
@@ -881,73 +872,74 @@ var _ = Describe("[kube-api]cluster installation", func() {
 		}, "2m", "2s").Should(Equal(0))
 	})
 
-	It("deploy None platform CD with ACI and agents - wait for ready, delete CD and verify ACI and agents deletion", func() {
```
Comment (rollandf) [inline subsystem/kubeapi_test.go:884]: Why change an existing test and not add a new one?

File: subsystem/kubeapi_test.go [hunk 3]
```diff
@@ -479,28 +494,6 @@ func getDefaultAgentClusterInstallSpec(clusterDeploymentName string) *hiveext.Ag
 	}
 }
 
-func getDefaultNonePlatformAgentClusterInstallSpec(clusterDeploymentName string) *hiveext.AgentClusterInstallSpec {
```
Comment (lranjbar) [inline subsystem/kubeapi_test.go:482]: Why is this being removed?

File: subsystem/kubeapi_test.go (hunk 2 above)
Comment (slaviered) [reply subsystem/kubeapi_test.go:884]: +1

File: subsystem/kubeapi_test.go [hunk 4]
```diff
@@ -881,73 +894,75 @@ var _ = Describe("[kube-api]cluster installation", func() {
...
 		Eventually(func() bool {
-			aci := &hiveext.AgentClusterInstall{}
-			err := kubeClient.Get(ctx, installkey, aci)
-			return apierrors.IsNotFound(err)
-		}, "30s", "10s").Should(Equal(true))
-		By("Verify ClusterDeployment Agents were deleted")
-		Eventually(func() int {
-			return len(getClusterDeploymentAgents(ctx, kubeClient, clusterKey).Items)
-		}, "2m", "2s").Should(Equal(0))
+			dbCluster, err := common.GetClusterFromDBWhere(db, common.UseEagerLoading, common.IncludeDeletedRecords, "kube_key_name = ? and kube_key_namespace = ?", clusterkey.Name, clusterkey.Namespace)
```
Comment (eranco74) [inline subsystem/kubeapi_test.go:943]: `clusterkey` isn't defined in this `It` scope
That's probably why the test fails to find the cluster in the DB...

Comment (isaacdorfman) [reply subsystem/kubeapi_test.go:943]: clusterkey := types.NamespacedName{
			Namespace: Options.Namespace,
			Name:      clusterDeploymentSpec.ClusterInstallRef.Name,
		}

File: subsystem/kubeapi_test.go [hunk 5]
```diff
@@ -950,6 +966,86 @@ var _ = Describe("[kube-api]cluster installation", func() {
...
+			Namespace: Options.Namespace,
+			Name:      ignitionTokenSecretName,
+		}
+		err := kubeClient.Update(ctx, agent)
+		Expect(err).To(BeNil())
+
+		By("Verify Ignition Token in DB")
+		Eventually(func() bool {
+			dbHost := GetHostByKubeKey(ctx, db, hostkey, waitForReconcileTimeout)
+			GinkgoLogger(fmt.Sprintf("ignition endpoint :\n%#v", dbHost.IgnitionEndpointToken))
```
Comment (rollandf) [inline subsystem/kubeapi_test.go:1044]: You can remove the log

**Pull request merged**